tests/json_schema_test_suite_test.rs

```diff
@@ -23,7 +28,33 @@ use stringreader::StringReader;
     "optional_cross_draft_.*",
     "optional_ecmascript_regex_.*",
     "refRemote_.*", // no.
-    "ref_.*", // TODO: make reference handling robust
+    // TODO: make reference handling robust
+    // The following are references that aren't yet handled by js2n (remote URIs, ref to something
+    // else than properties or definitions, etc.)
+    "ref_0_3.*", // reference to the whole schema `#` not yet supported
+    "ref_12_1.*", // reference to bare URI `node` (no fragment, no leading slash). Should fail
+                  // because invalid, but js2n replace that by a `Dyn` contract
+    "ref_13_0.*", // schemars doesn't properly percent-decode URIs
+    "ref_14_1.*", // reference to an anchor "#foo" not yet supported
+    "ref_15_1.*", // anchor + remote URI
+    "ref_16_1.*", // external reference (remote URI)
+    "ref_18_2.*", // external reference (local file URI)
+    "ref_19_2.*", // external reference (local file URI),
+    "ref_20_1.*", // external reference (remote URI)
+    "ref_21_1.*", // urn:uuid URI scheme not supported
+    "ref_26_1.*", // urn:uuid URI scheme not supported
+    "ref_27_1.*", // urn:uuid URI scheme not supported,
+    "ref_28_0.*", // external reference (remote URI)
+    "ref_29_0.*", // external reference (remote URI)
+    "ref_2_1.*", // internal ref to an array element (#/items/0)
+    "ref_30_0.*", // external reference (remote URI)
+    "ref_31_1.*", // external reference (absolute path /absref/foobar.json)
+    "ref_34_1.*", // non top-level definition ("#/definitions//definitions/")
+    "ref_3_5.*", // schemars doesn't properly percent-decode URIs
+    "ref_5_1.*", // not related to external ref, but js2n doesn't properly ignore other components
+                 // when the `ref` field is used
```

Comment:
I guess this one should be fixed soon, as it's an actual bug in json-schema-to-nickel, not just a missing feature.

Reply:
Yes. I wanted to proceed with this PR anyway, but I agree. I just currently don't know the JSON schema spec enough to be sure of the exact rule to follow, but I just notice that the example schema has both `$ref` set and a array schema that says that length must be at most 2, the ref is just a general array schema, and the example has 3 elements and must be passing. So it really feels like what's tested is that the initial array schema must be ignored.

Reply:
Then let's proceed the way it is, and figure out the JSON schema spec later :+1: